### ISSUE TITLE
exit code 143 on graceful shutdown

### ISSUE THREAD

**Issue by gaerfield** [neo4j/docker-neo4j#30]
I'm starting and stopping neo4j with a docker-compose-file:
```
version: '2'
services:
  neo4j:
    image: neo4j:2.3.2
    ports:
      - 7474:7474
    volumes:
      - ./data:/data
```
On `docker-compose stop` I'm getting an '143' exit-code, when the log-messages seems to be fine (console-output from windows docker toolbox):
```
Stopping neo4j_neo4j_1 ...
neo4j_1 | 2016-03-21 09:00:52.599+0000 INFO  Neo4j Server shutdown initiated by request
neo4j_1 | 2016-03-21 09:00:52.643+0000 INFO  Successfully shutdown Neo4j Server
neo4j_1 | 2016-03-21 09:00:52.725+0000 INFO  Successfully stopped database
Stopping neo4j_neo4j_1 ... done
neo4j_neo4j_1 exited with code 143
```

**Comment by benbc** [comment]
Thanks @gaerfield. That's certainly not expected, we'll take a look.

**Comment by benbc** [comment]
@gaerfield I've looked into this and I think it's working as expected. Docker stops containers by sending them a `SIGTERM`. Following standard behaviour Neo4j reports this by exiting with `143`. The same thing happens if you issue a `docker stop` to a container running outside `docker-compose` (although `docker-compose` makes it more obvious by logging the exit code).

I'm going to close this now, but please reopen if you think my analysis is wrong.

**Comment by gaerfield** [comment]
Thanks for the info, this is good to know. I thought, ``docker stop`` is going to initiate a "normal" shutdown, which should result in an ExitCode ``0``. I was wrong here. Thank you for checking.

**Comment by spacecowboy** [comment]
@gaerfield this will be fixed in `3.2` 

See related PR: https://github.com/neo4j/neo4j/pull/9108
